algorithms/src/srs/universal_prover.rs

```diff
@@ -12,11 +12,21 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use crate::{
+    fft::domain::{FFTPrecomputation, IFFTPrecomputation},
+    polycommit::sonic_pc::CommitterKey,
+};
 use snarkvm_curves::PairingEngine;
 
 /// `UniversalProver` is used to compute evaluation proofs for a given commitment.
-#[derive(Clone, Debug, Default, PartialEq, Eq)]
+#[derive(Clone, Debug, PartialEq, Eq)]
 pub struct UniversalProver<E: PairingEngine> {
+    /// The committer key for the underlying KZG10 scheme.
+    pub committer_key: CommitterKey<E>,
```

Comment:
There should be some method included in this struct that allows parameter fetching (or checked injection perhaps if developers store their own powers of tau - say for offline proofs) in a web friendly way. This effectively means an asynchronous parameter fetching path. Ditto for any constructor methods for the `UniversalProver` within other structs that has this as a member.